### PR TITLE
Improved JIT support for torch.einsum

### PR DESCRIPTION
**Pull request by heitorschueroff**
Added JIT support for the vararg version of `torch.einsum`. Note that JIT does not support the Python's Ellipsis object (`...`)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1741,22 +1741,20 @@ def check(fn, jitted, *args):
...
-        with self.assertRaises(RuntimeError):
-            torch.jit.script(sublist_format)
-
         x = make_tensor((5,), 'cpu', torch.float32)
         y = make_tensor((10,), 'cpu', torch.float32)
 
-        check(equation_format, torch.jit.script(equation_format), x, y)
-        check(equation_format, torch.jit.trace(equation_format, (x, y)), x, y)
-        check(sublist_format, torch.jit.trace(sublist_format, (x, y)), x, y)
+        for fn in [equation_format, equation_format_varargs, sublist_format]:
```
Comment (mruberry) [inline test/test_jit.py:1755]: Nice programmatic enumeration

**Comment by mruberry** [comment]
This is a really cool extension of einsum jit support. @Krovatkin, is there someone familiar with the jit's varargs handling who can take a look from the jit side?

**Comment by heitorschueroff** [comment]
@Krovatkin any updates here? @Lilyjjo and @eellison could one of you (or both) take a look at this PR please?

**Comment by Lilyjjo** [comment]
Hi, I'm really not familiar with the varargs -- if @eellison isn't either I can help find someone on the JIT team who is!

File: test/test_jit.py [hunk 2]
```diff
@@ -1740,22 +1740,20 @@ def check(fn, jitted, *args):
             self.assertEqual(fn(*args), jitted(*args))
 
         def equation_format(x, y):
-            return torch.einsum('i,j->ij', (x, y))
+            return torch.einsum('i,j->ij', [x, y])
```
Comment (Krovatkin) [inline test/test_jit.py:1743]: why was this changed?

**Review by Krovatkin — COMMENTED** [review]
looks good overall! I have a quick question above.

File: test/test_jit.py (hunk 2 above)
Comment (heitorschueroff) [reply test/test_jit.py]: No reason, I changed it back.

**Review by Krovatkin — APPROVED** [review]
:shipit: 

Looks gut to me on the JIT side @mruberry
